app/services/pipeline_report_service.rb

```diff
@@ -249,13 +249,15 @@ def get_pipeline_status(pipeline_run)
       }
     end
 
+    # pipeline_run.results_finalized? indicates that the results monitor sees all outputs are in a finished state
+    # (either loaded or failed). This can be true if there were errors, so we still need to check pipeline_run.failed? as well.
+    # pipeline_run.report_ready? indicates if taxon_counts output is loaded and available to use in report generation.
+    # This is only true if no errors have occurred for taxon_counts.
     pipeline_status = "WAITING"
-    # pipeline_run.completed? and pipeline_run.finalized? both indicate if the pipeline run
-    # is no longer being checked by the pipeline monitor (i.e. the run has completed),
-    # but .completed? includes extra logic for an old version of pipeline without run stages.
-    if pipeline_run.completed?
+    if pipeline_run.results_finalized?
       pipeline_status = "COMPLETE"
-    elsif pipeline_run.failed?
+    end
+    if pipeline_run.failed?
```

Comment:
So, can it be finalized AND failed? 
If so, we should do `if failed elsif finalized` instead. Seems easier to understand.

Reply:
Yes, it can be both finalized and failed. Changed to `failed/elsif finalized`!